erpnext/controllers/selling_controller.py

```diff
@@ -548,3 +548,48 @@ def update_project_billing_and_sales(self):
 			doc.set_gross_margin(update=True)
 			doc.set_status(update=True)
 			doc.notify_update()
+
+	@frappe.whitelist()
+	def set_rate_as_cost(self):
+		for item in self.items:
+			if item.item_code:
+				cost_rate = get_rate_as_cost(item.item_code, item.warehouse, item.batch_no, item.serial_no,
+					self.posting_date, self.posting_time, item.qty, self.doctype, self.name,
+					self.company, item.delivery_note, item.delivery_note_item)
+				item.rate = cost_rate
+				item.discount_percentage = 0
+				item.margin_rate_or_amount = 0
+
+		self.calculate_taxes_and_totals()
+
+
+@frappe.whitelist()
+def get_rate_as_cost(item_code, warehouse, batch_no, serial_no, posting_date, posting_time, qty, voucher_type,
```

Comment:
Make this a controller method as well with `self` and `item` object arguments to simplify the arguments.
Also remove whitelist since this doesn't do everything a standalone API should